### PR TITLE
feat(docs): add default theme, CSS Variables, and CSS-in-JS

### PR DESCRIPTION
**Pull request by 0618**
#### Description of changes

- ~~**theming/default-theme** to be add to sidebar~~ remove **theming/default-theme** and will add it in another [ticket](https://app.asana.com/0/1201736086077838/1202286620749761/f)
- **theming/alternative-styling** will be **guides/css-in-js**
- move css-variables from authenticator to **theming/css-variables**

|   |  Before  | After   |
|---|---|---|
| ~~Default Theme~~  | ~~ image~~  | ~~ image~~ |
| Alternative Styling  | image  |  image  |
| CSS Variables  | image  |  image  |

#### Issue #, if available

https://app.asana.com/0/1201736086077838/1202286620749764/f

#### Description of how you validated changes

manually tested locally

#### Checklist

- [x] PR description included
- [x] `yarn test` passes
- [ ] Tests are updated
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated
- [ ] Relevant documentation is changed or added (and PR referenced)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by reesscot — COMMENTED** [review]
Actually, @dbanksdesign was thinking of moving this page to guides and naming it CSS in JS. Can you please move it to guides instead?

* Move to path `/guides/css-in-js`
* Change name to CSS in JS in Nav and move "CSS in JS" heading to frontmatter title.

File: docs/src/data/meta.ts [hunk 1]
```diff
@@ -252,8 +252,8 @@ export const META_INFO: MetaInfo = {
     title: 'Guides',
     description: 'How to protect routes with Amplify Authenticator.',
   },
-  '/theming/default-theme': {
-    title: 'Theming - Default Theme',
+  '/guides/css-in-js': {
+    title: 'Guides - CSS in JS',
     description:
       'A Default Theme is a structured collection of the default design decisions that change the appearance of a UI library. An Amplify UI theme is a structured object of design tokens, breakpoints, and overrides. - Amplify UI',
```
Comment (0618) [inline docs/src/data/meta.ts:258]: How should we update the description?

**Comment by 0618** [comment]
> Actually, @dbanksdesign was thinking of moving this page to guides and naming it CSS in JS. Can you please move it to guides instead?
> 
> * Move to path `/guides/css-in-js`
> * Change name to CSS in JS in Nav and move "CSS in JS" heading to frontmatter title.

Done! (any suggestions on how to update the meta-description?)

File: docs/src/data/meta.ts [hunk 2]
```diff
@@ -267,8 +267,13 @@ export const META_INFO: MetaInfo = {
...
-    title: 'Theming - Alternative Styling',
+  '/theming/css-variables': {
+    title: 'Theming - CSS Variables',
+    description:
+      'CSS Variables used in Amplify UI. You can override these values to update the look and feel.',
+  },
+  '/guides/css-in-js': {
+    title: 'Guides - CSS in JS',
     description:
       'One of the goals of Amplify is to be a good citizen of the environment it exists in. As such, Amplify works hard to not interfere with other tools that are being used. - Amplify UI',
```
Comment (reesscot) [inline docs/src/data/meta.ts:278]:
```suggestion
      'Amplify UI supports using many popular CSS-in-JS frameworks to override styling on our components.',
```

File: docs/src/data/meta.ts [hunk 3]
```diff
@@ -267,8 +267,13 @@ export const META_INFO: MetaInfo = {
     description:
       'Dark Mode uses a darker color palette for all screens, making foreground content stand out against darker backgrounds. ',
   },
-  '/theming/alternative-styling': {
-    title: 'Theming - Alternative Styling',
+  '/theming/css-variables': {
+    title: 'Theming - CSS Variables',
+    description:
+      'CSS Variables used in Amplify UI. You can override these values to update the look and feel.',
```
Comment (reesscot) [inline docs/src/data/meta.ts:273]:
```suggestion
      'CSS Variables used in Amplify UI. You can override these values to update the look and feel of your application.',
```

File: docs/next.config.js [hunk 4]
```diff
@@ -72,7 +72,7 @@ module.exports = withNextPluginPreval({
       },
       {
         source: '/ui/theme/alternativeStyling',
-        destination: '/theming/alternative-styling',
```
Comment (reesscot) [inline docs/next.config.js:75]: I think we also need a redirect from `/theming/alternative-styling` to new url `/guides/css-in-js'

File: docs/src/components/Layout/SecondaryNav.tsx [hunk 5]
```diff
@@ -101,12 +101,8 @@ export const SecondaryNav = (props) => {
...
             Dark mode
           </NavLink>
-          <NavLink
-            {...props}
-            platforms={['react', 'vue', 'angular']}
-            href="/theming/alternative-styling"
-          >
-            Alternative styling
+          <NavLink {...props} href="/theming/css-variables">
+            CSS Variables
```
Comment (reesscot) [inline docs/src/components/Layout/SecondaryNav.tsx:105]: This doesn't apply to Flutter, shouldn't we add the platforms?
```
             platforms={['react', 'vue', 'angular']}
```

**Review by reesscot — CHANGES_REQUESTED** [review]
Just a few more changes

File: docs/src/components/Layout/SecondaryNav.tsx (hunk 5 above)
Comment (0618) [reply docs/src/components/Layout/SecondaryNav.tsx:109]: good catch! Done.

File: docs/next.config.js (hunk 4 above)
Comment (0618) [reply docs/next.config.js:75]: Good catch! Done

**Review by dbanksdesign — APPROVED** [review]

File: docs/next.config.js [hunk 6]
```diff
@@ -58,48 +58,53 @@ module.exports = withNextPluginPreval({
       {
         source: '/ui/primitives/stepperField',
         destination: '/components/stepperfield',
-        permanent: false,
+        permanent: true,
```
Comment (reesscot) [inline docs/next.config.js:61]: good catch!

File: docs/src/pages/[platform]/theming/default-theme/colors.tsx [hunk 7]
```diff
@@ -1,33 +0,0 @@
...
-  'orange',
-  'yellow',
-  'green',
-  'teal',
-  'blue',
-  'purple',
-  'pink',
-];
-
-export const Colors = () => {
```
Comment (reesscot) [inline docs/src/pages/[platform]/theming/default-theme/colors.tsx:16]: Looks like someone started building out the default-theme and just never put it on a page?

**Review by reesscot — APPROVED** [review]
shipit

File: docs/src/pages/[platform]/theming/default-theme/colors.tsx (hunk 7 above)
Comment (0618) [reply docs/src/pages/[platform]/theming/default-theme/colors.tsx:16]: yea, looks like it's not used anywhere 😂

**Pull request merged**